omc3/model/model_creators/sps_model_creator.py

```diff
@@ -42,21 +44,42 @@ def check_accelerator_instance(self) -> None:
         accel.verify_object()  # should have been done anyway, but cannot hurt (jdilly)
 
         # Creator specific checks
-        if accel.model_dir is None:
+        if accel.acc_model_path is None:
             raise AcceleratorDefinitionError(
                 "The accelerator definition is incomplete: "
-                "Model directory (outputdir option) was not given."
+                "SPS model creation only works with acc-models, but `acc-model-path` was not set. "
             )
         
-        if accel.acc_model_path is None:
+        # vvv ---  same as in LHC, maybe merge? (jdilly, 2025)
+        if accel.model_dir is None:
             raise AcceleratorDefinitionError(
                 "The accelerator definition is incomplete: "
-                "SPS model creation only works with acc-models, but `acc-model-path` was not set. "
+                "Model directory (outputdir option) was not given."
+            )
+
+        if accel.modifiers is None or not len(accel.modifiers):
+            raise AcceleratorDefinitionError(
+                "The accelerator definition is incomplete: No modifiers could be found. "
+                "Hint: If the accelerator class was instantiated from a model dir, "
+                f"make sure {JOB_MODEL_MADX_NOMINAL} exists and contains the '{MODIFIER_TAG}' tag."
```

Comment:
For me to be sure: we also consider something like the `opticsfile` / `strength file` to be a modifier right? This way it makes sense to have that error

Reply:
Yes. Basically everything that changes the parameters, once the sequence is built, we consider as modifiers. At least this is how it was in the LHC, where the optics files were always "modifiers". It also makes it easier from the gui to not having to deal with different flags for different machines. 
Sadly, the implementation of PS and PSB does not follow this philosophy.

Reply:
Got it, definitely agree it is better this way.